iron-core/src/main/java/io/axway/iron/core/internal/StoreManagerFactoryBuilderConfigurator.java

```diff
@@ -0,0 +1,201 @@
+            Supplier<?> builder = instantiateBuilder(config.supplierClass(), properties, config.baseName());
+            buildAndAssign(builder, entry.getKey(), storeManagerFactoryBuilder);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<Type, BuilderImplConfig> findComponentBuilders(Properties properties) {
+        Map<Type, BuilderImplConfig> map = new HashMap<>();  // ComponentBuilder -> property name base
+        for (String propertyName : properties.stringPropertyNames()) {
+            String className = getProperty(properties, propertyName);
+            if (className != null) {
+                try {
+                    Class clazz = Class.forName(className);
+                    if (Supplier.class.isAssignableFrom(clazz)) {
+                        Type type = ((ParameterizedTypeImpl) clazz.getGenericInterfaces()[0]).getActualTypeArguments()[0];
+                        if (map.containsKey(type)) {
+                            LOG.warn("A {} is already configured (with {})", type.getTypeName(), map.get(type).supplierClass().getName());
+                        }
+                        BuilderImplConfig builderImplConfig = new BuilderImplConfig(clazz, propertyName);
+                        LOG.info("Adding the configured class in the builder list: {{}: {}}", type.getTypeName(), className);
+                        map.put(type, builderImplConfig);
+                    }
+                } catch (ClassNotFoundException e) {
+                    // nothing to do
+                }
+            }
+        }
+        return map;
+    }
+
+    private Supplier<?> instantiateBuilder(Class<Supplier<?>> componentBuilderClazz, Properties properties, String baseName) {
+        try {
+            Constructor<Supplier<?>> constructor = componentBuilderClazz.getConstructor();
+            Supplier<?> builder = constructor.newInstance();
+
+            for (Method method : componentBuilderClazz.getDeclaredMethods()) {
+                if (method.getName().startsWith("set")) {
+                    Class<?>[] parameterTypes = method.getParameterTypes();
+                    if (parameterTypes.length == 1) {
+                        String propertyName = method.getName().substring(3);
+                        Object value = getPropertyValue(parameterTypes[0], properties, baseName, propertyName);
+                        if (value != null) { // do not call setter if value is null
+                            method.invoke(builder, value);
+                        } else if (isMandatory(method)) {
+                            throw new ConfigurationException("Parameter '" + propertyName + "' for '"+ componentBuilderClazz.getClass().getCanonicalName()+"' is mandatory!");
+                        }
+                    } else {
+                        LOG.error("The configured builder has setters with more than one parameter {{}.{}}", componentBuilderClazz.getName(), method.getName());
```

Comment:
why not an exception ?

Reply:
Initially I added the log so I can debug. But you're right, I'll throw an exception!